### ISSUE TITLE
compose: improving suggestion messages around `azd infra synth`

### ISSUE THREAD

**Issue by JeffreyCA** [Azure/azure-dev#4582]
Currently, if you add a new resource using `azd add` to an azd project that already has an `infra` folder, the following message is shown to the user after accepting the changes to `azure.yaml`:

> Run 'azd infra synth' to re-synthesize the infrastructure, then run 'azd provision' to provision these changes anytime later.

From my own experience encountering this message for the first time, I didn't really know what terms like "re-synthesize the infrastructure" or "overwrite with the synthesized versions" meant at first (I later learned these were common terminology used in IaC, Terraform, AWS CDK):

![Image](https://github.com/user-attachments/assets/ef77a233-9de2-4dd7-b23c-5b06aa0ef9d5)

I found the description used in the help text for `azd infra synth` to be more helpful:
```
Available Commands
  synth : Write IaC for your project to disk, allowing you to manage it by hand. (Beta)
```

Do you think instead of "re-synthesize the infrastructure" we could have similar wording to above? I think new users, especially those unfamiliar with IaC/Terraform lingo could benefit.

Also wondering if something like a more commonly used verb like `gen` or `generate` would make more sense instead of `synth` here.

**Comment by weikanglim** [comment]
Just curious, @JeffreyCA, in your mind, does "generate" sound better here?

**Comment by JeffreyCA** [comment]
> Just curious, [@JeffreyCA](https://github.com/JeffreyCA), in your mind, does "generate" sound better here?

Yes, "generate" sounds better to me, and maybe also that these are infrastructure *files* (on disk)

**Comment by puicchan** [comment]
@JeffreyCA - do we still need this or we can close with recent UX improvement and renaming of `infra synth`?

**Comment by JeffreyCA** [comment]
> [@JeffreyCA](https://github.com/JeffreyCA) - do we still need this or we can close with recent UX improvement and renaming of `infra synth`?

Thanks for the reminder - no this can be closed now 🙂
